readthedocs/doc_builder/base.py

```diff
@@ -89,16 +91,12 @@ def clean(self, **__):
 
     def docs_dir(self, docs_dir=None, **__):
         """Handle creating a custom docs_dir if it doesn't exist."""
-        checkout_path = self.project.checkout_path(self.version.slug)
         if not docs_dir:
             for doc_dir_name in ['docs', 'doc', 'Doc', 'book']:
-                possible_path = os.path.join(checkout_path, doc_dir_name)
+                possible_path = os.path.join(self.cwd, doc_dir_name)
                 if os.path.exists(possible_path):
-                    docs_dir = possible_path
-                    break
-        if not docs_dir:
-            docs_dir = checkout_path
-        return docs_dir
+                    return possible_path
+        return docs_dir or self.cwd
```

Comment:
This my be written with a first if checking for `docs_dir` and immediately return, to avoid having the `or` at the end and make it clearer.

```python
if docs_dir:
  return docs_dir

for ...
  if os.path.exists():
    return possible_path

return self.cwd
```